packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx

```diff
@@ -128,54 +106,148 @@ export const TransactionReviewModalContent = ({
-                    deviceModelInternal={deviceModelInternal}
-                    deviceUnitColor={device?.features?.unit_color}
-                    successText={<Translation id="TR_CONFIRMED_TX" />}
-                    onCancel={onCancel}
-                />
-            }
-        >
-            <TransactionReviewSummary
-                estimateTime={estimateTime}
-                tx={precomposedTx}
-                account={account}
-                network={network}
-                broadcast={precomposedForm.options.includes('broadcast')}
-                detailsOpen={detailsOpen}
-                onDetailsClick={() => setDetailsOpen(!detailsOpen)}
-                stakeType={stakeType}
-                actionText={getTransactionReviewModalActionText({
-                    stakeType,
-                    isRbfAction,
-                })}
-            />
-            <TransactionReviewOutputList
-                account={account}
-                precomposedForm={precomposedForm}
-                precomposedTx={precomposedTx}
-                signedTx={serializedTx}
-                decision={decision}
-                detailsOpen={detailsOpen}
-                outputs={outputs}
-                buttonRequestsCount={buttonRequestsCount}
-                isRbfAction={isRbfAction}
-                actionText={getTransactionReviewModalActionText({
-                    stakeType,
-                    isRbfAction,
-                    isSending,
-                })}
-                isSending={isSending}
-                setIsSending={() => setIsSending(true)}
-                stakeType={stakeType || undefined}
+        <NewModal.Backdrop>
+            <ConfirmOnDevice
+                title={<Translation id="TR_CONFIRM_ON_TREZOR" />}
+                steps={outputs.length + 1}
+                activeStep={serializedTx ? outputs.length + 2 : buttonRequestsCount}
+                deviceModelInternal={deviceModelInternal}
+                deviceUnitColor={device?.features?.unit_color}
+                successText={<Translation id="TR_CONFIRMED_TX" />}
+                onCancel={onCancel}
             />
-            <TransactionReviewEvmExplanation account={account} stakeType={stakeType} />
-        </StyledModal>
+            <NewModal.ModalBase
+                heading={<Translation id={areDetailsVisible ? 'TR_DETAIL' : actionLabel} />}
+                onBackClick={
+                    areDetailsVisible
+                        ? () => {
+                              setAreDetailsVisible(!areDetailsVisible);
+                          }
+                        : undefined
+                }
+                description={
+                    !areDetailsVisible && (
+                        <TransactionReviewSummary
+                            tx={precomposedTx}
+                            account={account}
+                            broadcast={precomposedForm.options.includes('broadcast')}
```

Comment:
I think this can be shortened to this, its already used above.

```suggestion
                            broadcast={isBroadcastEnabled}
```